From 20e5db556760b9d5749ff567781c94d25a810080 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Wed, 12 Apr 2023 11:01:19 +0000
Subject: [PATCH 15/23] [compiler-rt] [test] [profile] Skip a MSVC-style linker
 parameter in mingw mode

This one isn't essential for the execution of this test, so just
skip it when running tests in mingw mode.
---
 compiler-rt/test/profile/Windows/coverage-linkage.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/compiler-rt/test/profile/Windows/coverage-linkage.cpp b/compiler-rt/test/profile/Windows/coverage-linkage.cpp
index 648b3fb73a5d..315c5333aba4 100644
--- a/compiler-rt/test/profile/Windows/coverage-linkage.cpp
+++ b/compiler-rt/test/profile/Windows/coverage-linkage.cpp
@@ -3,7 +3,7 @@
 // RUN: env LLVM_PROFILE_FILE=%t.profraw %run %t
 // RUN: llvm-profdata show %t.profraw --all-functions | FileCheck %s
 
-// RUN: %clang_profgen -fcoverage-mapping -ffunction-sections -Wl,-opt:ref %s -o %t
+// RUN: %clang_profgen -fcoverage-mapping -ffunction-sections %if target={{.*-windows-msvc.*}} %{ -Wl,-opt:ref %} %s -o %t
 // RUN: env LLVM_PROFILE_FILE=%t.profraw %run %t
 // RUN: llvm-profdata show %t.profraw --all-functions | FileCheck %s
 
-- 
2.34.1

